python/tk_multi_publish2/api/tree.py

```diff
@@ -338,15 +343,31 @@ class _PublishTreeEncoder(json.JSONEncoder):
     """
     Implements the json encoder interface for custom publish tree serialization.
     """
-    def default(self, publish_tree):
-        return publish_tree.to_dict()
+    def default(self, data):
+        if isinstance(data, PublishTree):
+            return data.to_dict()
+        elif isinstance(data, sgtk.Template):
+            return {
+                "__tk_type": "sgtk.Template",
```

Comment:
lots of sgtk here. How about a simpler setup: `{_sgtk_custom_type: template, name: foo}` no need to have both `tk` and `sgtk` and i think nice if we can get the word custom in too.

Reply:
I've changed `__tk_type` to `_sgtk_custom_type`, but I'd rather keep the `sgtk.` bit in front of Template. Call it paranoia, but at least this way there's no confusion about what template we're talking about here and is a bit more future proof. Chances are we're never going to look at this line of code in the future anyway.